tests/popular_repos/popular_repos.bzl

```diff
@@ -0,0 +1,165 @@
+            "go/gcimporter15:go_default_xtest", # TODO(#417)
+            "refactor/importgraph:go_default_xtest", # TODO(#417)
+            "refactor/rename:go_default_test", # TODO(#417)
+            "cmd/guru/testdata/src/referrers:go_default_xtest", # Not a real test
+            "cmd/guru/testdata/src/referrers:go_default_test", # Not a real test
+            "container/intsets:go_default_xtest", # TODO(#413): External test depends on symbols defined in internal test.
+        ],
+    ),
+
+    dict(
+        name = "org_golang_google_grpc",
+        importpath = "google.golang.org/grpc",
+        commit = "3f10311ccf076b6b7cba28273df3290d42e60982",
+
+        # GRPC has already-generated protobuf definitions, and we don't currently
+        # register any protobuf toolchains in this WORKSPACE.  As such, the build
+        # should fail if we try to generate protobuf rules, but succeed if we
+        # disable generation.
+        build_file_proto_mode = "disable",
+        excludes = [
+            "test:go_default_test",
+            "examples/route_guide/mock_routeguide:go_default_xtest",
+            "examples/helloworld/mock_helloworld:go_default_xtest",
+            "credentials:go_default_test",
+            ":go_default_test",
+        ],
+    ),
+
+    dict(
+        name = "com_github_mattn_go_sqlite3",
+        importpath = "github.com/mattn/go-sqlite3",
+        commit = "83772a7051f5e30d8e59746a9e43dfa706b72f3b",
+        excludes = [],
+    ),
+  ]
+
+def _popular_repo_test_suites_impl(ctx):
+  ctx.file("WORKSPACE", """workspace(name="popular_repos")""")
+  content = ""
+  names = []
+  for repo in POPULAR_REPOS:
+    name = repo["name"]
+    names.append(name)
+    tests = ctx.execute(["cat", ctx.path(Label("@io_bazel_rules_go//tests/popular_repos:" + name))]).stdout.split("\n")
+    excludes = repo.get("excludes", [])
+    if ctx.os.name.startswith('windows'):
+      excludes = excludes + repo.get("windows_excludes", [])
+    elif ctx.os.name.startswith('mac'):
+      excludes = excludes + repo.get("darwin_excludes", [])
+    else:
+      excludes = excludes + repo.get(ctx.os.name + "_excludes", [])
+    excludes = ["@{}//{}".format(name, l) for l in excludes]
+    invalid_excludes = [t for t in excludes if not t in tests]
+    if invalid_excludes:
+      fail("Invalid excludes found: {}".format(invalid_excludes))
+    content += """
+test_suite(
+    name = "{name}",
+    tests = {tests},
+)
+""".format(name=name, tests=[t for t in tests if t and not t in excludes])
+  content += """
+test_suite(
+    name = "all",
+    tests = {tests},
+)
+""".format(tests=names)
+
+  ctx.file("BUILD.bazel", content)
+
+
+_popular_repo_test_suites = repository_rule(
+    implementation = _popular_repo_test_suites_impl,
+)
+
+def popular_repos():
+  names = []
+  for repo in POPULAR_REPOS:
+    names.append(repo["name"])
+    repo = {k:v for k,v in repo.items() if not "excludes" in k}
+    go_repository(**repo)
```

Comment:
Some of these repositories (`org_golang_x_tools`, proto stuff) will override dependencies of rules_go. This will only affect rules_go, but it could cause some weirdness. Let's wrap these declarations in `_maybe` to avoid this.